x/incentive/keeper/keeper.go

```diff
@@ -39,10 +45,267 @@ func NewKeeper(
+			// --------------------(Cosmos SDK staking module implementation)------------------------
+			// del.Shares = val.GetDelegatorShares().MulInt(amt).QuoInt(val.GetTokens())
+			// amt = del.Shares * val.GetTokens / val.GetDelegatorShares()
+			/****************************************************************************************/
+			amt := del.GetShares().MulInt(val.GetTokens()).Quo(val.GetDelegatorShares()).TruncateInt()
+
+			// Sum the individual delegation
+			delegatedAmt.Add(amt)
+			return false
+		},
+	)
+
+	return delegatedAmt
+}
+
+// Find out active incentive params
+func (k Keeper) FindProperIncentiveParm(ctx sdk.Context, epochIdentifier string) (bool, types.IncentiveInfo, types.IncentiveInfo) {
+	// Incentive params initialize
+	stakeIncentive := types.IncentiveInfo{}
+	lpIncentive := types.IncentiveInfo{}
+
+	// Current block timestamp
+	timestamp := ctx.BlockTime().Unix()
+	foundIncentive := false
+
+	// Fetch incentive params
+	params := k.GetParams(ctx)
+
+	// Find approporiate Incentive Info
+	// Consider epochIdentifier and start time
+	// Consider epochNumber as well
+	for _, ii := range params.StakeIncentives {
+		if ii.EpochIdentifier == epochIdentifier && timestamp >= ii.StartTime.Unix() {
+			stakeIncentive = ii
+			foundIncentive = true
+			break
+		}
+	}
+
+	// Find approporiate Incentive Info
+	for _, ii := range params.LpIncentives {
+		if ii.EpochIdentifier == epochIdentifier && timestamp >= ii.StartTime.Unix() {
+			lpIncentive = ii
+		}
+	}
+
+	// return found, stake, lp incentive params
+	return foundIncentive, stakeIncentive, lpIncentive
+}
+
+// Update uncommitted token amount
+// Called back through epoch hook
+func (k Keeper) UpdateUncommittedTokens(ctx sdk.Context, epochIdentifier string) {
+	// Recalculate total committed info
+	k.UpdateTotalCommitmentInfo(ctx, epochIdentifier)
+
+	// Iterate all delegations for the specified delegator
+	// Process to increase uncomitted token amount of Eden & Eden boost
+	k.cmk.IterateCommitments(
+		ctx, epochIdentifier,
+		func(commitments ctypes.Commitments) bool {
+			// Find out active incentive params
+			foundIncentive, stakeIncentive, lpIncentive := k.FindProperIncentiveParm(ctx, epochIdentifier)
+
+			// If we don't have incentive params ready
+			if !foundIncentive {
+				return true
+			}
+
+			// Commitment owner
+			creator := commitments.Creator
+
+			// Calculate delegated amount per delegator
+			delegatedAmt := k.CalculateDelegatedAmount(ctx, creator)
+
+			// Calculate new uncommitted Eden tokens for LP, staker, and Eden token holders
+			new_uncommitted_eden_tokens := k.CalculateNewUncommittedEdenTokens(ctx, delegatedAmt, commitments, stakeIncentive, lpIncentive)
+
+			// Calculate new uncommitted Eden-Boost tokens for staker and Eden token holders
+			new_uncommitted_eden_boost_tokens := k.CalculateNewUncommittedEdenBoostTokens(ctx, delegatedAmt, commitments, epochIdentifier, stakeIncentive, lpIncentive)
```

Comment:
I think golang does not use snake case variable name but will need to use camel case mostly. `newUncommittedEdenBoostTokens`